force-app/main/default/aura/nksChatConversationHandler/nksChatConversationHandlerController.js

```diff
@@ -29,36 +29,38 @@
-        const chatToolkit = component.find('chatToolkit');
-        const eventRecordId = event.getParam('recordId');
-        const workspace = component.find('workspace');
-        const eventFullID = helper.convertId15To18(eventRecordId);
+        const type = event.getParam('type');
+        if (type === 'sessionEnded') {
+            const eventRecordId = event.getParam('recordId');
+            const workspace = component.find('workspace');
+            const chatToolkit = component.find('chatToolkit');
 
-        workspace
-            .getAllTabInfo()
-            .then((res) => {
-                const eventTab = res.find((content) => content.recordId === eventFullID);
-                if (!eventTab) return;
-                helper.setTabColor(workspace, eventTab.tabId, 'success');
-            })
-            .catch(() => {
-                //Errors require manual handling.
-            });
+            workspace
+                .getAllTabInfo()
+                .then((tabInfoList) => {
+                    const eventTab = tabInfoList.find((tab) => tab.recordId === eventRecordId);
+                    if (eventTab) {
+                        // eslint-disable-next-line @lwc/lwc/no-async-operation, @locker/locker/distorted-window-set-timeout
+                        setTimeout(() => {
+                            helper.setTabColor(workspace, eventTab.tabId, 'success');
+                        }, 1000);
+                    }
+                })
+                .catch((error) => {
+                    console.error('Error retrieving tab info for setting color:', error);
+                });
 
-        chatToolkit
-            .getChatLog({
-                recordId: eventRecordId
-            })
-            .then((result) => {
-                let conversation = result.messages;
-                let filteredConversation = conversation.filter(function (message) {
-                    //Filtering out all messages of type supervisor and AgentWhisper as these are "whispers" and should not be added to the journal
-                    return message.type !== 'Supervisor' && message.type !== 'AgentWhisper';
+            chatToolkit
+                .getChatLog({ recordId: eventRecordId })
+                .then((chatLog) => {
+                    const filteredConversation = chatLog.messages.filter(
+                        (message) => message.type !== 'Supervisor' && message.type !== 'AgentWhisper'
+                    );
+                    helper.callStoreConversation(component, filteredConversation, eventRecordId);
+                })
+                .catch((error) => {
+                    console.error('Error retrieving chat log:', error);
                 });
-                helper.callStoreConversation(component, filteredConversation, eventRecordId);
-            })
-            .catch(() => {
-                //Errors require manual handling.
-            });
+        }
```

Comment:
Detta funke vel ikke? chatToolkit.getChatLog() gjør vel ingenting for Messaging.